### PR TITLE
Converts build-api-markdown-documenter.yml pipeline to 1ES template

### PR DESCRIPTION
**Pull request by seanimam**
## Description

This PR converts build-api-markdown-documenter.yml to the new 1ES template required by Microsoft with the minimal required set of template changes required. This new template enables security and auditing checks by Microsoft. Read more on how the logic for converting to a 1ES pipeline works on [the official 1ES wiki](https://eng.ms/docs/cloud-ai-platform/devdiv/one-engineering-system-1es/1es-docs/1es-pipeline-templates/onboarding/overview)

A temporary 1ES folder was created for converted pipeline templates. This enables other pipelines to remain unchanged and use the original templates while new PR's are opened for individually converted other pipelines to 1ES as required by Microsoft. **You can easily view what the template changes are by looking at the diff in [the following PR](https://github.com/microsoft/FluidFramework/pull/19244) which is a fork of this branch and simply moves the template changes from the 1ES folder to the original so there is a clear diff present on github**

- Once all pipelines are moved over, we can start moving the files out of the 1ES folder and eventually remove the folder since it will no longer have any files (This is after we convert the rest of the required pipelines). 

## Reviewer Guidance

- Confirm the newly converted pipelines run successfully. Look at the completed steps and compare with a recent successful run on main to ensure the expected steps run (plus the new SDL and 1ES security checks)
- Make sure all the relevant templates and pipelines for build-api-markdown-documenter.yml have been converted
- Make sure there are no unnecessary changes or 1ES template and/or pipeline conversions
- Make sure the yml is valid -- a successful pipeline run should suffice for confirming this

### CODE REVIEW

**Comment by alexvy86** [comment]
For clarity, let's update the description; it's making reference to repo-policy-check.yml, and asks to check build-client and build-azure for review, where it should be the one for api-markdown-documenter.

**Comment by seanimam** [comment]
> For clarity, let's update the description; it's making reference to repo-policy-check.yml, and asks to check build-client and build-azure for review, where it should be the one for api-markdown-documenter.

Updated the description.

**Review by alexvy86 — APPROVED** [review]
🚀

File: tools/pipelines/build-api-markdown-documenter.yml [hunk 1]
```diff
@@ -57,17 +57,25 @@ pr:
...
     - tools/pipelines/templates/include-use-node-version.yml
     - tools/pipelines/templates/include-process-test-results.yml
     - scripts/*
 
+variables:
+  - template: /tools/pipelines/templates/include-vars.yml@self
+    parameters:
+      publishOverride: '${{ parameters.publishOverride }}'
+      releaseBuildOverride: '${{ parameters.releaseBuildOverride }}'
+      buildNumberInPatch: ''
```
Comment (Josmithr) [inline tools/pipelines/build-api-markdown-documenter.yml:73]: Sorry, got to this a bit late. For future PRs, when we have empty arguments like this, I would recommend adding a quick comment explaining what it is / why we don't want it or want the default.

File: tools/pipelines/templates/1ES/include-policy-check.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+# Licensed under the MIT License.
+
+# include-policy-check
+#
+# This template can be included in pipelines that run check scripts as part of their pipeline. The check scripts are
+# executed serially in a separate stage from the main build.
```
Comment (Josmithr) [inline tools/pipelines/templates/1ES/include-policy-check.yml:7]: Nit (for the future): Could we expand the documentation to explain what we mean by "checks"?

Comment (Josmithr) [reply tools/pipelines/templates/1ES/include-policy-check.yml:7]: Actually, after reading through the code, it "checks" seems pretty open-ended. It seems like the purpose of this template is really just to run a series of npm scripts after installing node (and then verifying that nothing changed, git-wise). This feels much more general than "policy-check". I wonder if we have other templates that have essentially the same structure that we could deduplicate.

Comment (Josmithr) [reply tools/pipelines/templates/1ES/include-policy-check.yml:7]: These are future considerations though :)

**Pull request merged**